src/MoveUntilTouchTopicController.cpp

```diff
@@ -0,0 +1,216 @@
+  }
+
+  // load name of force/torque sensor handle from paramter
+  std::string ft_wrench_name;
+  if (!nh.getParam("forcetorque_wrench_name", ft_wrench_name)) {
+    ROS_ERROR("Failed to load 'forcetorque_wrench_name' parameter.");
+    return false;
+  }
+  // load name of force/torque tare handle from paramter
+  std::string ft_tare_name;
+  if (!nh.getParam("forcetorque_tare_name", ft_tare_name)) {
+    ROS_ERROR("Failed to load 'forcetorque_tare_name' parameter.");
+    return false;
+  }
+
+  // load force/torque saturation limits from parameter
+  if (!nh.getParam("sensor_force_limit", mForceLimit)) {
+    ROS_ERROR("Failed to load 'sensor_force_limit' paramter.");
+    return false;
+  }
+  if (!nh.getParam("sensor_torque_limit", mTorqueLimit)) {
+    ROS_ERROR("Failed to load 'sensor_torque_limit' paramter.");
+    return false;
+  }
+  if (mForceLimit < 0) {
+    ROS_ERROR("sensor_force_limit must be positive or zero");
+    return false;
+  }
+  if (mTorqueLimit < 0) {
+    ROS_ERROR("sensor_torque_limit must be positive or zero");
+    return false;
+  }
+
+  // subscribe to sensor data
+  forceTorqueDataSub = nh.subscribe(ft_wrench_name, 1000, &MoveUntilTouchTopicController::forceTorqueDataCallback, this);
+
+  // action client to kick off taring
+  tareActionClient = std::unique_ptr<TareActionClient>(new TareActionClient(nh, ft_tare_name));
+
+  // start action server
+  mFTThresholdActionServer.reset(
+      new actionlib::ActionServer<SetFTThresholdAction>{
+          nh, "set_forcetorque_threshold",
+          std::bind(&MoveUntilTouchTopicController::setForceTorqueThreshold, this,
+                    std::placeholders::_1),
+          false});
+  mFTThresholdActionServer->start();
+
+  // initialize base trajectory controller
+  return initController(robot, nh);
+}
+
+//=============================================================================
+void MoveUntilTouchTopicController::forceTorqueDataCallback(const geometry_msgs::WrenchStamped& msg)
+{
+  mForce.x() = msg.wrench.force.x;
+  mForce.y() = msg.wrench.force.y;
+  mForce.z() = msg.wrench.force.z;
+  mTorque.x() = msg.wrench.torque.z;
+  mTorque.y() = msg.wrench.torque.z;
+  mTorque.z() = msg.wrench.torque.z;
+}
+
+//=============================================================================
+void MoveUntilTouchTopicController::taringTransitionCallback(TareActionClient::GoalHandle goalHandle) {
+  if (goalHandle.getResult()) {
+    if (goalHandle.getResult()->success) {
+      ROS_INFO("Taring completed!");
+      mTaringCompleted.store(true);
+    }
+  }
+}
+
+//=============================================================================
+void MoveUntilTouchTopicController::starting(const ros::Time& time)
+{
+  // start asynchronous tare request
+  ROS_INFO("Starting Taring");
+  pr_control_msgs::TriggerGoal goal;
+  mTareGoalHandle = tareActionClient->sendGoal(goal, boost::bind(&MoveUntilTouchTopicController::taringTransitionCallback, this, _1));
```

Comment:
Nit: I think we now prefer `std::bind`.

Reply:
`SimpleActionClient`'s function expects a `boost::function`, so I don't think we can easily switch to `std`, can we?